File: invesdwin-scripting-parent/invesdwin-scripting-julia/invesdwin-scripting-julia-runtime-contract/src/test/java/de/invesdwin/context/julia/runtime/contract/callback/ParametersAndReturnsTestLong.jl
println("getLong")
if isdefined(Main, :getLong) && !isnothing(getLong)
	error("getLong already defined!")
end
getLong = callback("getLong")
println(typeof(getLong))
println(getLong)
if typeof(getLong) != Int64
	error("getLong not Int64!")
end
callback("setLong",getLong)

println("getLongVector")
if isdefined(Main, :getLongVector) && !isnothing(getLongVector)
	error("getLongVector already defined!")
end
getLongVector = callback("getLongVector")
println(eltype(getLongVector))
println(getLongVector)
if eltype(getLongVector) != Int64
	error("getLongVector not Int64!")
end
callback("setLongVector",getLongVector)

println("getLongVectorAsList")
if isdefined(Main, :getLongVectorAsList) && !isnothing(getLongVectorAsList)
	error("getLongVectorAsList already defined!")
end
getLongVectorAsList = callback("getLongVectorAsList")
println(eltype(getLongVectorAsList))
println(getLongVectorAsList)
if eltype(getLongVectorAsList) != Int64
	error("getLongVectorAsList not Int64!")
end
callback("setLongVectorAsList",getLongVectorAsList)

println("getLongMatrix")
if isdefined(Main, :getLongMatrix) && !isnothing(getLongMatrix)
	error("getLongMatrix already defined!")
end
getLongMatrix = callback("getLongMatrix")
println(eltype(getLongMatrix))
println(getLongMatrix)
if eltype(getLongMatrix) != Int64
	error("getLongMatrix not Int64!")
end
callback("setLongMatrix",getLongMatrix)

println("getLongMatrixAsList")
if isdefined(Main, :getLongMatrixAsList) && !isnothing(getLongMatrixAsList)
	error("getLongMatrixAsList already defined!")
end
getLongMatrixAsList = callback("getLongMatrixAsList")
println(eltype(getLongMatrixAsList))
println(getLongMatrixAsList)
if eltype(getLongMatrixAsList) != Int64
	error("getLongMatrixAsList not Int64!")
end
callback("setLongMatrixAsList",getLongMatrixAsList)

File: invesdwin-scripting-parent/invesdwin-scripting-runtime-lua/src/test/java/de/invesdwin/scripting/runtime/lua/tests/callback/ParametersAndReturnsTestLong.lua
arraysToString = java.method(java.import('de.invesdwin.util.collections.Arrays'),'toString','java.lang.Object')

print("getLong")
if(getLong ~= nil) then
	error("getLong already defined!")
end
getLong = callback("getLong")
getLongType = type(getLong)
print(getLongType)
print(getLong)
if(getLongType ~= "number") then
	error("getLong not Long!")
end
callback("setLong", getLong)

print("getLongVector")
if(getLongVector ~= nil) then
	error("getLongVector already defined!")
end
getLongVector = callback("getLongVector")
getLongVectorType = type(getLongVector[1])
print(getLongVectorType)
print(arraysToString(getLongVector))
if(getLongVectorType ~= "number") then
	error("getLongVector not Long!")
end
callback("setLongVector", getLongVector)

print("getLongVectorAsList")
if(getLongVectorAsList ~= nil) then
	error("getLongVectorAsList already defined!")
end
getLongVectorAsList = callback("getLongVectorAsList")
getLongVectorAsListType = type(getLongVectorAsList[1])
print(getLongVectorAsListType)
print(arraysToString(getLongVectorAsList))
if(getLongVectorAsListType ~= "number") then
	error("getLongVectorAsList not Long!")
end
callback("setLongVectorAsList", getLongVectorAsList)

print("getLongMatrix")
if(getLongMatrix ~= nil) then
	error("getLongMatrix already defined!")
end
getLongMatrix = callback("getLongMatrix")
getLongMatrixType = type(getLongMatrix[1][1])
print(getLongMatrixType)
print(arraysToString(getLongMatrix))
if(getLongMatrixType ~= "number") then
	error("getLongMatrix not Long!")
end
callback("setLongMatrix", getLongMatrix)

print("getLongMatrixAsList")
if(getLongMatrixAsList ~= nil) then
	error("getLongMatrixAsList already defined!")
end
getLongMatrixAsList = callback("getLongMatrixAsList")
getLongMatrixAsListType = type(getLongMatrixAsList[1][1])
print(getLongMatrixAsListType)
print(arraysToString(getLongMatrixAsList))
if(getLongMatrixAsListType ~= "number") then
	error("getLongMatrixAsList not Long!")
end
callback("setLongMatrixAsList", getLongMatrixAsList)
